lib/config_verify/util.go

```diff
@@ -0,0 +1,306 @@
+package config_verify
+
+import (
+	"container/list"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/chef/automate/lib/config_parser"
+	"github.com/chef/automate/lib/stringutils"
+)
+
+const (
+	FOR_CERTS = " for certs "
+)
+
+func validateAutomateFQDN(fqdn string) error {
+	if strings.Contains(fqdn, " ") {
+		return errors.New("domain name cannot contain spaces")
+	}
+
+	// Check for "http://" or "https://" in the FQDN value
+	if strings.HasPrefix(fqdn, "http://") || strings.HasPrefix(fqdn, "https://") {
+		return fmt.Errorf("fqdn should not include protocol (http:// or https://)")
+	}
+	return nil
+}
+
+func validateChannel(channel string) (bool, string) {
+	return validateOneOf("Channel", channel, []string{"current", "acceptance", "dev"})
+}
+
+func validateUpgradeStrategy(strategy string) (bool, string) {
+	return validateOneOf("Upgrade strategy", strategy, []string{"none", "at-once"})
+}
+
+func validatePackageCleanupMode(mode string) (bool, string) {
+	return validateOneOf("Package cleanup mode", mode, []string{"conservative", "aggressive", "disabled"})
+}
+
+func validateOneOf(msgPrefix string, input string, allowedValues []string) (bool, string) {
+	valid := stringutils.SliceContains(allowedValues, input)
+	if !valid {
+		return valid, msgPrefix + " must be one of: " + strings.Join(allowedValues, ", ")
+	}
+	return valid, ""
+}
+
+func checkIPAddress(ip string) error {
+	if net.ParseIP(ip) == nil {
+		return errors.New("ip Address is invalid")
+	} else {
```

Comment:
we can directly return without else block.

Reply:
Yes.